micrometer-tracing/src/main/java/io/micrometer/tracing/Span.java

```diff
@@ -370,6 +371,37 @@ public Span start() {
          */
         Builder startTimestamp(long startTimestamp, TimeUnit unit);
 
+        /**
+         * Adds a link to the newly created {@code Span}.
+         *
+         * <p>
+         * Links are used to link {@link Span}s in different traces. Used (for example) in
+         * batching operations, where a single batch handler processes multiple requests
+         * from different traces or the same trace.
+         * @param traceContext the context of the linked {@code Span}.
+         * @return this
+         * @since 1.1.0
+         */
+        default Builder addLink(TraceContext traceContext) {
```

Comment:
If links are used to link `Span`s, should not this receive a Span instead of the context? Internally it can fetch the context from the Span but the user could do:
```java
span.addLink(otherSpan);
```
instead of
```java
span.addLink(otherSpan.context());
```

WDYT?